airflow/providers/hashicorp/secrets/vault.py

```diff
@@ -58,155 +54,127 @@ class VaultBackend(BaseSecretsBackend, LoggingMixin):
     :type mount_point: str
+    :param kv_engine_version: Select the version of the engine to run (``1`` or ``2``, default: ``2``)
+    :type kv_engine_version: int
     :param token: Authentication token to include in requests sent to Vault.
         (for ``token`` and ``github`` auth_type)
     :type token: str
-    :param kv_engine_version: Select the version of the engine to run (``1`` or ``2``, default: ``2``)
-    :type kv_engine_version: int
     :param username: Username for Authentication (for ``ldap`` and ``userpass`` auth_type)
     :type username: str
     :param password: Password for Authentication (for ``ldap`` and ``userpass`` auth_type)
     :type password: str
-    :param role_id: Role ID for Authentication (for ``approle`` auth_type)
+    :param key_id: Key ID for Authentication (for ``aws_iam`` and ''azure`` auth_type)
+    :type key_id: str
+    :param secret_id: Secret ID for Authentication (for ``approle``, ``aws_iam`` and ``azure`` auth_types)
+    :type secret_id: str
+    :param role_id: Role ID for Authentication (for ``approle``, ``aws_iam`` auth_types)
     :type role_id: str
     :param kubernetes_role: Role for Authentication (for ``kubernetes`` auth_type)
     :type kubernetes_role: str
-    :param kubernetes_jwt_path: Path for kubernetes jwt token (for ``kubernetes`` auth_type, deafult:
+    :param kubernetes_jwt_path: Path for kubernetes jwt token (for ``kubernetes`` auth_type, default:
         ``/var/run/secrets/kubernetes.io/serviceaccount/token``)
     :type kubernetes_jwt_path: str
-    :param secret_id: Secret ID for Authentication (for ``approle`` auth_type)
-    :type secret_id: str
     :param gcp_key_path: Path to GCP Credential JSON file (for ``gcp`` auth_type)
+           Mutually exclusive with gcp_keyfile_dict
     :type gcp_key_path: str
+    :param gcp_keyfile_dict: Dictionary of keyfile parameters. (for ``gcp`` auth_type).
+           Mutually exclusive with gcp_key_path
+    :type gcp_keyfile_dict: dict
     :param gcp_scopes: Comma-separated string containing GCP scopes (for ``gcp`` auth_type)
     :type gcp_scopes: str
+    :param azure_tenant_id: Tenant of azure (for ``azure`` auth_type)
+    :type azure_tenant_id: str
+    :param azure_resource: Resource if of azure (for ``azure`` auth_type)
+    :type azure_resource: str
+    :param radius_host: Host for radius (for ``radius`` auth_type)
+    :type radius_host: str
+    :param radius_secret: Secret for radius (for ``radius`` auth_type)
+    :type radius_secret: str
+    :param radius_port: Port for radius (for ``radius`` auth_type)
+    :type radius_port: str
     """
     def __init__(  # pylint: disable=too-many-arguments
         self,
         connections_path: str = 'connections',
         variables_path: str = 'variables',
         url: Optional[str] = None,
         auth_type: str = 'token',
-        mount_point: str = 'secret',
+        mount_point: Optional[str] = None,
```

Comment:
This would need a note in Updating.md unless we already define `secret` as default `mount_point` for `token`, `auth_type`. And since we still have `auth_type: str = 'token'` , do you think we should keep it as it is ?

Reply:
Actually that was misleading in the original implementation. I checked the paths and it was not working correctly. If you look at the past implementation mount_point was not passed to the auth methods:

For example: 
```
_client.auth_kubernetes(role=self.kubernetes_role, jwt=jwt)
```

when you look inside, mount_point defaults to "kubernetes". So no matter what you specified as mount_point, always "kubernetes" was used. There was actually no way to authenticate with the selected mount point in the earlier version - you only retrieve the secret with the mount_point but authentication was always done with the default "kubernetes" one. 

But you are right that this might need an UPDATING entry. I actually start thinking that maybe it is too much of a change (though I believe that this is desired behaviour to keep same mount_point in both "auth" and "get_secret" case)

WDYT?

Reply:
The default mount point is different for each Secret Engine. In Secrets Backend for Vault we only supported getting Secrets from KV Secrets Engine and its default mount_point is `secret` -> https://hvac.readthedocs.io/en/v0.10.3/source/hvac_api_secrets_engines.html#hvac.api.secrets_engines.KvV2.read_secret_version - The default mount point for reading secret from KV Secret Engine is `secret` (also https://learn.hashicorp.com/vault/kubernetes/secret-store-driver#set-a-secret-in-vault). 

https://github.com/apache/airflow/blob/1c9374d2573483dd66f5c35032e24140864e72c0/airflow/providers/hashicorp/secrets/vault.py#L200-L206

We did support different auth_methods but didn't support multiple Secrets Engine in 1.10.10

Reply:
I added  better explanation in th docstring and UPDATING.md entry

Reply:
@kaxil -> yeah, It was one engine, that's fine. But the problem is that the mount_point is also parameter in the "auth" method and the "secret" was never passed there even if it was set. So the sequence it was working so far when you passed "secret" as mount point was:

1) Vault Backend was created:
```
      vb = new VaultBackend(auth_type="kubernetes" ) 
      assert (vb.mount_point == "secret") 
```

So far so good. We have VB with "kubernetes" auth_type and "secret" mount_point.


2) Then when "client()" method was called, this happened:

```
_client.auth_kubernetes(role=self.kubernetes_role, jwt=jwt)
```

When you look at the implementation of the "auth_kubernetes" method. It actually uses "kubernetes" as  the default mount_point:

https://github.com/hvac/hvac/blob/develop/hvac/v1/__init__.py#L1623

```
   def auth_kubernetes(self, role, jwt, use_token=True, mount_point='kubernetes'):
        """POST /auth/<mount_point>/login

        :param role: Name of the role against which the login is being attempted.
```

3) Then when read_secret() was called it was called with  "secret" mount_point:

```
 if self.kv_engine_version == 1: 
     response = self.client.secrets.kv.v1.read_secret( 
         path=secret_path, mount_point=self.mount_point 
     ) 
 else: 
     response = self.client.secrets.kv.v2.read_secret_version( 
         path=secret_path, mount_point=self.mount_point) 
```

So as the result - the "auth_kubernetes" was called with "kubernetes" mount_point and "get_secret" with "secret" mount_point. Which I think was wrong. I believe both should be called with the same mount point.

My change makes it consistent - both "auth" and "get_secret" methods will be always called with the same mount_point (either default for the authentication method or the one specified in the configuration).

Reply:
But this PR too only supports getting Secrets from a KV Engine, isn't it ?

Reply:
>Having multiple auth methods enables you to use an auth method that makes the most sense for your use case of Vault and your organization.

https://www.vaultproject.io/docs/auth#auth-methods

Reply:
TBH since I haven't used any other Secrets Engine I am not sure 🤷 . 

I have used and tested the KV Secrets Engine, both v1 and v2 and it is pretty straightforward but haven't used the other ones

Reply:
Here is one tutorial I found: https://medium.com/google-cloud/vault-auth-and-secrets-on-gcp-51bd7bbaceb

Reply:
Also, this one is nice: https://www.hashicorp.com/resources/fleetsmith-deploys-vault-gcp/

We might want to try it out but yeah we can have a separate parameter for `auth_mount_point`

Reply:
OK I think I understand now. I will change it back to what it was but I will update description to clarify that the mount point is only used for secrets retrieval!

Reply:
Fixed. It should be backwards compatible now - yet you can still provide custom auth_mount_point :)

Reply:
Mount point shouldn't be optional should it? It's a required setting

Reply:
It was set below in init. I moved it up to the constructor parameter now.